master/internal/rm/kubernetesrm/pod.go

```diff
@@ -344,28 +342,28 @@ func (p *pod) deleteKubernetesResources(ctx *actor.Context) {
 	}
 
 	ctx.Log().Infof("requesting to delete kubernetes resources")
-	ctx.Tell(p.resourceRequestQueue, deleteKubernetesResources{
-		handler:       ctx.Self(),
-		namespace:     p.namespace,
-		podName:       p.podName,
-		configMapName: p.configMapName,
-	})
+	p.resourceRequestQueue.deleteKubernetesResources(
+		p.resourceErrorCtx(ctx),
+		p.namespace,
+		p.podName,
+		p.configMapName,
+	)
 
 	p.resourcesDeleted = true
 }
 
-func (p *pod) receiveResourceCreationCancelled(ctx *actor.Context) {
-	ctx.Log().Infof("pod actor notified that resource creation was canceled")
+func (p *pod) receiveResourceCreationFailed(ctx *actor.Context, err resourceCreationFailed) {
+	p.syslog.WithError(err).Error("pod actor notified that resource creation failed")
+	p.insertLog(ctx, time.Now().UTC(), err.Error())
+}
+
+func (p *pod) receiveResourceCreationCancelled() {
+	p.syslog.Info("pod creation canceled")
 	p.resourcesDeleted = true
```

Comment:
wouldn't this bool technically need locking now?

Reply:
good point, replaced it with an atomic bool